src/TestCase.js

```diff
@@ -121,4 +184,11 @@ export class TestCase {
 	getDimensions() {
 		return this.dimensions;
 	}
-}
\ No newline at end of file
+
+	/**
+	 * @param {TestCaseFailedState|TestCaseFinishedState|TestCaseIsRunningState} state
+	 */
+	updateState( state ) {
```

Comment:
The internal methods could call this too instead of manually setting the variable.